### PR TITLE
[TEST] Refactor test classes to match multi frontend protocol

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

As Kyuubi supports REST frontend protocol now, and MySQL frontend protocol is coming soon, Thrift is not the only one, this PR refacotor the test classes to match the changes.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/test/scala/org/apache/kyuubi/WithSecuredDFSService.scala [hunk 1]
```diff
@@ -59,5 +59,5 @@ trait WithSecuredDFSService extends KerberizedTestHelper {
     super.afterAll()
   }
 
-  def getHadoopConf(): Configuration = miniDFSService.getHadoopConf
+  def getHadoopConf: Configuration = miniDFSService.getHadoopConf
```
Comment (pan3793) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/WithSecuredDFSService.scala:62]: Remove parentheses because the function has no side effects.

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/RestFrontendService.scala [hunk 2]
```diff
@@ -33,7 +33,7 @@ import org.apache.kyuubi.service.{AbstractFrontendService, Serverable, Service}
  * A frontend service based on RESTful api via HTTP protocol.
  * Note: Currently, it only be used in the Kyuubi Server side.
  */
-private[server] class RestFrontendService(override val serverable: Serverable)
+class RestFrontendService(override val serverable: Serverable)
```
Comment (pan3793) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/RestFrontendService.scala:36]: Maybe `RESTfulFrontendService` is a better name? cc @yanghua

Comment (yanghua) [reply kyuubi-server/src/main/scala/org/apache/kyuubi/server/RestFrontendService.scala:36]: sounds good. +1

**Comment by pan3793** [comment]
any thoughts? @yaooqinn

**Comment by yaooqinn** [comment]
https://docs.github.com/en/rest

**Comment by yaooqinn** [comment]
https://developer.twitter.com/en/docs

**Comment by yaooqinn** [comment]
https://developer.paypal.com/docs/api/overview/#make-rest-api-calls

**Comment by pan3793** [comment]
Sound like we should change `rest` to `REST`?

**Comment by yaooqinn** [comment]
https://www.ics.uci.edu/~fielding/pubs/dissertation/rest_arch_style.htm

**Comment by yaooqinn** [comment]
> Sound like we should change `rest` to `REST`?

the `rest` is ok to me from the given URLs, they are mostly contains lower case `rest`

**Comment by yaooqinn** [comment]
I can not find any supporting evidence from your PR description so I did some research on my own. Some from the best practices of big companies and some from the original author.

**Comment by pan3793** [comment]
WDYT other changes of this PR?

File: kyuubi-common/src/test/scala/org/apache/kyuubi/service/NoopBackendService.scala [hunk 3]
```diff
@@ -21,6 +21,7 @@ import org.apache.kyuubi.KyuubiException
 import org.apache.kyuubi.session.{NoopSessionManager, SessionManager}
 
 class NoopBackendService extends AbstractBackendService("noop") {
+
```
Comment (yaooqinn) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/service/NoopBackendService.scala:24]: nit

**Review by yaooqinn — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master
